### PR TITLE
Make thrift_client_pool support tbinary protocol

### PR DESCRIPTION
**Pull request by shuzhang1989**
serverset_thrift_router will use it to talk to non-fbthrift servers.

### CODE REVIEW

**Comment by newpoo** [comment]
Can we implement it by adding a template parameter "bool USE_BINARY_PROTOCOL = false" to ThriftClientPool?

File: common/thrift_client_pool.h [hunk 1]
```diff
@@ -167,6 +175,10 @@ class ThriftClientPool {
         if (FLAGS_channel_send_timeout_ms > 0) {
           channel->setTimeout(FLAGS_channel_send_timeout_ms);
         }
+        if (binary_protocol) {
```
Comment (newpoo) [inline common/thrift_client_pool.h]: We can use USE_BINARY_PROTOCOL here directly. And don't need to change the interface for EventLoop.

File: common/thrift_client_pool.h [hunk 2]
```diff
@@ -226,10 +238,10 @@ class ThriftClientPool {
   };
 
  public:
+
   // Create a new pool of n_io_threads IO threads. The threads are owned by the
   // pool.
-  explicit ThriftClientPool(
-      const uint16_t n_io_threads =
+  explicit ThriftClientPool(const uint16_t n_io_threads =
```
Comment (newpoo) [inline common/thrift_client_pool.h]: Can you undo all the unnecessary changes including adding empty new lines.

File: common/thrift_client_pool.h [hunk 3]
```diff
@@ -29,8 +29,10 @@
 #include "folly/futures/Promise.h"
 #include "folly/io/async/EventBase.h"
 #include "folly/SocketAddress.h"
+#include "thrift/lib/cpp/transport/THeader.h"
```
Comment (newpoo) [inline common/thrift_client_pool.h]: can you move it one line lower to make them sorted.

**Comment by shuzhang1989** [comment]
@newpoo thanks for the review!
